presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunctionBuilder.java

```diff
@@ -249,6 +223,56 @@ public MethodsGroup build()
         }
     }
 
+    public static class ChoiceBuilder
+    {
+        private final Class<?> clazz;
+        private Signature signature;
+        private boolean nullableResult;
+        private List<ArgumentProperty> argumentProperties = emptyList();
+        private List<MethodsGroup> methodsGroups = new ArrayList<>();
+
+        private ChoiceBuilder(Class<?> clazz, Signature signature)
+        {
+            this.clazz = clazz;
+            this.signature = requireNonNull(signature, "signature is null");
+        }
+
+        public ChoiceBuilder implementation(Function<MethodsGroupBuilder, MethodsGroupBuilder> methodGroupSpecification)
```

Comment:
`methodGroupSpecification` -> `methodsGroupSpecification`?

Reply:
I am trying to understand your comment. The code is not clear?